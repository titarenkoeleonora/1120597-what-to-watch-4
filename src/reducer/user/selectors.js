import NameSpace from "../name-space.js";

export const getAuthorizationStatus = (state) => state[NameSpace.USER].authorizationStatus;

export const getAuthorizationInfo = (state) => state[NameSpace.USER].authorizationInfo;

export const getIsAuthorizing = (state) => state[NameSpace.USER].isAuthorizing;
