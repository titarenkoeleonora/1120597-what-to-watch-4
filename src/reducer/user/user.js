import {UserActionCreator} from "../actions/user-action-creator";
import {UserActionType} from "../actions/user-action-type";
import {extend} from "../../utils";
import createUserInfo from "../../adapter/user";

const AuthorizationStatus = {
  AUTH: `AUTH`,
  NO_AUTH: `NO_AUTH`,
};

const initialState = {
  authorizationStatus: AuthorizationStatus.NO_AUTH,
  isAuthorizing: false,
  authorizationInfo: {
    id: 0,
    email: ``,
    name: ``,
    avatar: ``,
  }};

const Operation = {
  checkAuth: () => (dispatch, getState, api) => {
    return api.get(`/login`)
      .then((response) => {
        dispatch(UserActionCreator.requireAuthorization(AuthorizationStatus.AUTH));
        dispatch(UserActionCreator.loadAuthorizationInfo(createUserInfo(response.data)));
      })
      .catch((err) => {
        throw err;
      });
  },

  login: (authData) => (dispatch, getState, api) => {
    return api.post(`/login`, {
      email: authData.login,
      password: authData.password,
    })
      .then((response) => {
        dispatch(UserActionCreator.requireAuthorization(AuthorizationStatus.AUTH));
        dispatch(UserActionCreator.loadAuthorizationInfo(createUserInfo(response.data)));
        dispatch(UserActionCreator.isNotAuthorizing());
      })
      .catch(() => {
        dispatch(UserActionCreator.requireAuthorization(AuthorizationStatus.NO_AUTH));
      });
  },
};


const reducer = (state = initialState, action) => {
  switch (action.type) {
    case UserActionType.REQUIRED_AUTHORIZATION:
      return extend(state, {
        authorizationStatus: action.payload,
      });
    case UserActionType.AUTHORIZATION_INFO:
      return extend(state, {
        authorizationInfo: action.payload
      });
    case UserActionType.IS_AUTHORIZING:
      return extend(state, {
        isAuthorizing: true,
      });
    case UserActionType.IS_NOT_AUTHORIZING:
      return extend(state, {
        isAuthorizing: false,
      });
  }

  return state;
};

export {reducer, initialState, AuthorizationStatus, Operation};
