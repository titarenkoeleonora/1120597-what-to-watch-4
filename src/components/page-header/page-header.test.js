import React from "react";
import renderer from "react-test-renderer";
import PageHeader from "./page-header";

import configureStore from "redux-mock-store";
import {Provider} from "react-redux";
import NameSpace from "../../reducer/name-space.js";
import {AuthorizationStatus} from "../../reducer/user/user";
import {Router} from "react-router-dom";
import history from "../../history";

const mockStore = configureStore([]);

it(`Correctly PageFooter render`, () => {
  const store = mockStore({
    [NameSpace.USER]: {
      authorizationStatus: AuthorizationStatus.NO_AUTH,
      isAuthorizing: false,
      authorizationInfo: {
        id: 0,
        email: ``,
        name: ``,
        avatar: ``,
      }},
  });

  const tree = renderer
    .create(
        <Router history={history}>
          <Provider store={store}>
            <PageHeader
              authorizationStatus={AuthorizationStatus.NO_AUTH}
              isAuthorizing={false}
              authorizationInfo={{
                id: 0,
                email: ``,
                name: ``,
                avatar: ``,
              }}
            />
          </Provider>
        </Router>
    );

  expect(tree).toMatchSnapshot();
});
