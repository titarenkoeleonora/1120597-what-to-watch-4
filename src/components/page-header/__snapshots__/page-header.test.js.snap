// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Correctly PageFooter render 1`] = `
<header
  className="page-header movie-card__head"
>
  \`   
  <div
    className="logo"
  >
    <a
      className="logo__link"
      href="main.html"
    >
      <span
        className="logo__letter logo__letter--1"
      >
        W
      </span>
      <span
        className="logo__letter logo__letter--2"
      >
        T
      </span>
      <span
        className="logo__letter logo__letter--3"
      >
        W
      </span>
    </a>
  </div>
  <div
    className="user-block"
  >
    <div
      className="user-block__avatar"
    >
      <img
        alt="User avatar"
        height="63"
        src="img/avatar.jpg"
        width="63"
      />
    </div>
  </div>
</header>
`;
